test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/ProjectSystems/LegacyPackageReferenceRestoreUtilityTests.cs

```diff
@@ -893,6 +893,143 @@ public async Task DependencyGraphRestoreUtility_LegacyPackageRef_Restore_Package
+                    File.Create(projectLockFilePath).Close();
+
+                    var restoreSummaries = await DependencyGraphRestoreUtility.RestoreAsync(
+                        testSolutionManager,
+                        dgSpec,
+                        restoreContext,
+                        new RestoreCommandProvidersCache(),
+                        (c) => { },
+                        sourceRepositoryProvider.GetRepositories(),
+                        Guid.Empty,
+                        false,
+                        true,
+                        testLogger,
+                        CancellationToken.None);
+
+                    foreach (var restoreSummary in restoreSummaries)
+                    {
+                        Assert.True(restoreSummary.Success);
+                        Assert.False(restoreSummary.NoOpRestore);
+                    }
+
+                    Assert.True(File.Exists(projectLockFilePath));
+
+                    // delete existing restore output files
+                    File.Delete(Path.Combine(vsProjectAdapter.MSBuildProjectExtensionsPath, "project.assets.json"));
+
+                    // clean packages folder
+                    Directory.Delete(testSolutionManager.GlobalPackagesFolder, true);
+                    Directory.CreateDirectory(testSolutionManager.GlobalPackagesFolder);
+
+                    // overwrite the package
+                    FileUtility.Delete(packageSource);
+                    var newPackageA = new SimpleTestPackageContext("packageA", "1.0.0");
+                    newPackageA.AddFile("lib/net45/a1.dll");
+                    var newPackageB = new SimpleTestPackageContext("packageB", "1.0.0");
+                    newPackageB.AddFile("lib/net45/b1.dll");
+                    SimpleTestPackageUtility.CreateOPCPackage(newPackageA, packageSource);
+                    SimpleTestPackageUtility.CreateOPCPackage(newPackageB, packageSource);
+
+                    // Act
+                    restoreSummaries = await DependencyGraphRestoreUtility.RestoreAsync(
+                        testSolutionManager,
+                        dgSpec,
+                        restoreContext,
+                        new RestoreCommandProvidersCache(),
+                        (c) => { },
+                        sourceRepositoryProvider.GetRepositories(),
+                        Guid.Empty,
+                        false,
+                        true,
+                        testLogger,
+                        CancellationToken.None);
+
+                    // Assert
+                    foreach (var restoreSummary in restoreSummaries)
+                    {
+                        Assert.False(restoreSummary.Success);
+                        Assert.True(restoreSummary.Errors.Count > 0);
+                        var nu1403Message = restoreSummary.Errors.FirstOrDefault(message => (message as RestoreLogMessage).Code == NuGetLogCode.NU1403);
```

Comment:
Shouldn't this be `SingleOrDefault(…)` instead of `FirstOrDefault(…)`?  There should be at most 1 NU1403.  If there were 2 this test wouldn't catch it.

Reply:
fixed.